Assignment3.txt

```diff
@@ -0,0 +1,8 @@
+Anya Lauria
+
+1. The code is invalid. This is because while the print statement is trying to output a variable which has not been 
+   called 
+2. There is a semicolon after the for statement 
+3. i. 0(n^2)
+  ii. 0(n^3)
```

Comment:
The answer is actually O(n^4) - again, for loops that run from 0 to n have the time complexity O(n). So the top loop is O(n), the middle Loop is O(n^2) iterations because it iterates from 1 to i^2. The if statement is O(1 / n) times because of the modulo function. Lastly, the inner loop  is O(n^2) iterations because it iterates from 0 to j and j = i^2 = n^2. 

Therefore, the sum = -(1/12)*n + (3/8)*n^2 - (5/12)*n^3 + (1/8)*n^4  = O(n^4)